feat: add id token support

drev74 opened the pull request.

evert commented:
Hey, I've rejected a very similar PR in the past, #118 

But I've actually changed my mind since then. I still don't intend to turn this in a full-blown OIDC library, but this seems like a fairly low-lift way for people to just get the idToken and potentially do the JWT parsing themselves, so well... lets do it I guess.

However, I have notes:

* This should have a test.
* Existing tests and Typescript should pass
* I think the type of `idToken` in OAuth2Token should be `idToken?: string` instead of `idToken: string | null`, and the token should ideally on the object (not even as an undefined value) so it doesn't appear as `null` and in serializations for users that never care about OIDC.

drev74 commented:
Hi @evert 

Thanks for your feedback. I think adding ID Token won't hurt this lib, since most if not all modern Oauth2 providers DO support OIDC :point_up: . In fact, I also own an OIDC/Oauth2 provider which returns ID token by default :hugs: 

Providing an ID token saves Frontend from making at least 1 extra call to the provider endpoint to get a `user ID` hence yields in a lower latency and a lower traffic load.

UPD:

I fixed my code as per your feedback and added a test. For some reason, current `eslint` config forces `double quotes`  in my VSCode so apart of tiny changes we have a larger formatting :roll_eyes:

evert commented:
> since most if not all modern Oauth2 providers DO support OIDC

There's lots of OAuth2 use-cases. You don't really see this as much for example when OAuth2 is used to integrate with APIs of commercial software, and frankly OIDC is overkill for many cases.

If you can, try to undo those formatting changes as they are causing a conflict, and will also down the road cause a conflict with #172 which is also actively being worked on!

drev74 commented:
Restored original formatting. Added a test for `id_token`

evert commented:
There's still a conflict and some changes unrelated to this PR! Once those are fix, this is good to merge, thanks!
